packages/angular/projects/ui-angular/src/lib/services/authenticator.service.ts

```diff
@@ -22,29 +29,25 @@ const logger = new Logger('state-machine');
 })
 export class AuthenticatorService implements OnDestroy {
   private _authState: AuthMachineState;
+  private _authStatus: AuthStatus = 'configuring';
   private _authService: AuthInterpreter;
   private _machineSubscription: Subscription;
   private _facade: ReturnType<typeof getServiceFacade>;
+  private _hubSubject: Subject<void>;
+  private _unsubscribeHub: () => void;
 
   constructor() {
     const machine = createAuthenticatorMachine();
+    this._authService = interpret(machine).start();
 
-    const authService = interpret(machine).start();
-
-    this._machineSubscription = authService.subscribe((state: unknown) => {
-      const newState = state as AuthMachineState;
-      this._authState = newState;
-      this._facade = getServiceFacade({
-        send: authService.send,
-        state: newState,
-      });
-    });
-
-    this._authService = authService;
+    this.setupMachineSubscription();
+    this.setupHubListner();
+    this.getInitialAuthStatus();
```

Comment:
Constructor was getting a bit large so I separated it to different functions. `setupHubListener` is the new logic. Note that their  order really don't matter.